onnxruntime/core/optimizer/gelu_fusion.cc

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/optimizer/initializer.h"
+#include "core/optimizer/gelu_fusion.h"
+#include "core/graph/graph_utils.h"
+#include "float.h"
+#include <deque>
+
+using namespace ONNX_NAMESPACE;
+using namespace ::onnxruntime::common;
+namespace onnxruntime {
+
+static bool CheckConstantInput(const Graph& graph, const NodeArg& input_arg, float expected_value) {
+  auto shape = input_arg.Shape();
+  auto dim_size = shape->dim_size();
+  if (dim_size != 0) {
+    // only check scalar.
+    return false;
+  }
+
+  const ONNX_NAMESPACE::TensorProto* tensor_proto = nullptr;
+  graph.GetInitializedTensor(input_arg.Name(), tensor_proto);
+  auto init_const = std::make_unique<Initializer>(tensor_proto);
+  auto data_type = *(input_arg.Type());
```

Comment:
Why does an Initializer need to be created? Can't we check the data type and read the value from the tensor_proto directly?

Reply:
because I want to reuse data read logic  wrapped within Initializer. If I read value directly from tensor_proto, need have logic checking value is raw data or non-raw data. Reimplementing this in a function would be not necessary. How do you think?